float32 magnitude
bool ignoreAngle
float32 angle
